packages/devtools-reps/src/object-inspector/index.js

```diff
@@ -196,12 +209,29 @@ class ObjectInspector extends Component {
     if (expand === true) {
       const {
         getObjectProperties,
+        getObjectEntries,
         loadObjectProperties,
+        loadObjectEntries,
       } = this.props;
 
       const value = getValue(item);
+      const parent = getParent(item);
+      const parentValue = getValue(parent);
+      const parentActor = parentValue
+        ? parentValue.actor
+        : null;
+
       if (nodeHasProperties(item) && value && !getObjectProperties(value.actor)) {
         loadObjectProperties(value);
+        return;
+      }
+
+      if (
+        nodeIsEntries(item)
+        && !hasAllEntriesInPreview(parent)
+        && !getObjectEntries(parentActor)
+      ) {
+        loadObjectEntries(parentValue);
```

Comment:
why differentiate between entries and properties?

also, do we want to return after loading?